include/nuttx/mtd/mtd.h

```diff
@@ -74,6 +74,9 @@
 #define MTDIOC_ERASESTATE _MTDIOC(0x000a) /* IN:  Pointer to uint8_t
                                            * OUT: Byte value that represents the
                                            *      erased state of the MTD cell */
+#define MTDIOC_FIRSTBLOCK _MTDIOC(0x000b) /* IN:  Pointer to off_t
```

Comment:
how about block device?

Reply:
How about add the new field to mtd_geometry_s or geometry? the start of block is also a key metric for block/mtd geometry.

Reply:
Does it make sense to have it with `mtd_geometry_s`? Looks a bit out of the scope of what geometry means here. If we added it to mtd_geometry_s, wouldn't be 0 for most cases?

Reply:
The mtd device is always zero, but partition on mtd isn't.

Reply:
block device also need similar change.

Reply:
> block device also need similar change.

I've just seen that block device partitions also handle `MTDIOC_GEOMETRY`, is this what you were referring to?
https://github.com/apache/incubator-nuttx/blob/fd19c36f5b7d36fb5b19946e035b5bff2d1b8628/fs/driver/fs_blockpartition.c#L249

Or do you mean to also add a new field to `struct geometry`?